quantmsio/core/quantms/msstats_in.py

```
@@ -0,0 +1,684 @@
+            print(f"Warning: Could not setup optimized processing: {e}")
+            # Fall back to original processing
+
+    def _detect_available_columns(self):
+        """Detect what columns are available in the report table."""
+        try:
+            # Try different methods to get column information
+            try:
+                columns_query = "PRAGMA table_info('report')"
+                columns_info = self._duckdb.execute(columns_query).df()
+                available_columns = set(columns_info["name"].tolist())
+            except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)